### PR TITLE
fix(queryinterface): decrement regression

### LINKED ISSUE #7810: Sequelize v4 does not update the instance when using .increment or .decrement
## What you are doing?

Hello, upgrading from v3 to v4, I had the following issue:

```js
const User = sequelize.define('user', {
  id: {
    type: DataTypes.UUID,
    primaryKey: true,
    defaultValue: DataTypes.UUIDV4
  },
  username: {
    type: DataTypes.STRING,
    allowNull: false
  },
  balance: {
    type: DataTypes.INTEGER,
    defaultValue: 0
  }
});

describe('.increment', ()=> {
  it('increments the property in the instance', () => {
    return User.create({username: 'alice'})
      .then(user => {
        user.increment('balance', 5)
        .then(()=> {
          expect(user.balance).toBe(5);
        });
      });
  });
});
```

## What do you expect to happen?
In Sequelize v3, the test would pass. the balance would be incremented in the object, without the need to reload it from the DB. After the upgrade, I expected the same behaviour as demonstrated by the test.

## What is actually happening?
With Sequelize v4, the record is updated in DB but the instance itself is not.
This is not documented in the upgrade notes.

__Sequelize version:__ v4.1.0

### PR DESCRIPTION
**Pull request by hieven**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [X] Did you follow the commit message conventions explained in CONTRIBUTING.md?

### Description of change

Basically just follow what query interface `increment` does

Closes #7810

### CODE REVIEW

**Comment by felixfbecker** [comment]
Is this a breaking change?

**Comment by hieven** [comment]
Hi @felixfbecker,
No, there is no breaking change between my initial PR and this fix PR.

In my initial PR, I just moved the original code in instance `increment` method to be class `increment` method and let instance method to call the class method.
(More details, please check #7394)

What I missed is that I didn't change the way how `decrement` in `query-interface` works, so it becomes a bug.

When someone uses instance method `decrement` under any dialect supports `returning`, the original instance won't get updated.

I also add another commit for testing this circumstance. Please help review it.

Thanks!
